### PR TITLE
Check for data field before checking for access token

### PR DESCRIPTION
**Pull request by a-roberts**
Should fix. https://github.com/tektoncd/dashboard/issues/1240

# Changes

Secrets can have no data, so check first

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by a-roberts** [comment]
Ready for a review please

image

**Comment by a-roberts** [comment]
```
(base) Adams-MBP:061 aroberts$ k get secret -o yaml | grep data | wc -l
      27
(base) Adams-MBP:061 aroberts$ k get secret | wc -l
      14
```

Something doesn't have data: (this is a default OCP 4.3 install) 👀 finding what

File: src/containers/Secrets/Secrets.js [hunk 1]
```diff
@@ -237,8 +237,10 @@ export /* istanbul ignore next */ class Secrets extends Component {
       });
     } else {
       secrets.forEach(secret => {
-        if (secret.data.accessToken) {
-          secretsToUse.push(secret);
+        if (secret.data) {
```
Comment (AlanGreene) [inline src/containers/Secrets/Secrets.js:240]: can we combine these and avoid the nesting? 🙏 

`if (secret.data && secret.data.accessToken) {`

Comment (a-roberts) [reply src/containers/Secrets/Secrets.js:240]: On it!

Comment (a-roberts) [reply src/containers/Secrets/Secrets.js]: Done - and the Prow PR just went in too so...happy days 🤞

**Comment by a-roberts** [comment]
Apparently we've gotta do a git commit amend force push, so let's confirm that 😸 

/retest

**Comment by a-roberts** [comment]
```
W0409 14:58:32.438] grouped write of manifest, lock and vendor: error while writing out vendor tree: failed to write dep tree: failed to export golang.org/x/sys: failed to fetch source for https://go.googlesource.com/sys: unable to get repository: Cloning into '/go/pkg/dep/sources/https---go.googlesource.com-sys'...
W0409 14:58:32.471] fatal: unable to access 'https://go.googlesource.com/sys/': gnutls_handshake() failed: The TLS connection was non-properly terminated.
```

hmm
/retest

**Comment by a-roberts** [comment]
Sooo... the empty PR passed. Let's reopen this and see if that does the trick

**Comment by a-roberts** [comment]
Last try before Boskos gets a break until Tuesday (for me)

/test pull-tekton-dashboard-integration-tests

**Pull request merged**